### PR TITLE
some graph enhancements and fixes

### PR DESCRIPTION
**Pull request by sepe81**
* fix: line chart for "group by days" must differentiate between yyyy-MM-dd in groupBy otherwise repeating days are summed up in following years
* fix: replace hungarian notation with better names (e.g. seriesArray, groupByString)
* fix typo: replace activites with activities
* enhancement: replace loop with reduce function (thanks to @CMRHDL)
* enhancement: restrict scope for several variables and use var for "-" and "."
* feature: include PolarArea vs. Pie chart (thanks to @CMRHDL)

### CODE REVIEW

**Comment by sepe81** [comment]
@inpercima please merge
@CMRHDL ready for optional review

File: src/main/resources/public/js/graphs.controller.js [hunk 1]
```diff
@@ -50,51 +46,57 @@
...
+        var groupByKey;
+        if (groupBy === 'year') {
+          label = readableDateLong[0];
+          groupByKey = label;
+        } else if (groupBy === 'month') {
+          label = readableDateLong[1] + dash + readableDateLong[0];
+          groupByKey = label;
+        } else if (groupBy === 'day') {
+          label = readableDateShort[2] + dot + readableDateShort[1];
+          groupByKey = label + dot + readableDateShort[0];
```
Comment (sepe81) [inline src/main/resources/public/js/graphs.controller.js]: this is important for the fix

**Comment by sepe81** [comment]
@inpercima Any more questions on that?

**Comment by inpercima** [comment]
Hey, was there a reason not to wait with this merge unti I checked this?

**Comment by sepe81** [comment]
I didn't want to wait any longer b/c the previous version was broken. I'm working on new features and don't want to fix too many merge conflicts. ;-)

**Pull request merged**